pkg/apis/core/v1alpha2/types.go

```diff
@@ -0,0 +1,76 @@
+// Copyright 2020 Antrea Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package v1alpha2
+
+import (
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// +genclient
+// +genclient:noStatus
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+type ExternalEntity struct {
+	metav1.TypeMeta `json:",inline"`
+	// Standard metadata of the object.
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+	// Desired state of the external entity.
+	Spec ExternalEntitySpec `json:"spec,omitempty"`
+}
+
+// ExternalEntitySpec defines the desired state for ExternalEntity.
+type ExternalEntitySpec struct {
+	// Endpoints is a list of external endpoints associated with this entity.
+	Endpoints []Endpoint `json:"endpoints,omitempty"`
+	// Ports maintain the list of named ports.
+	Ports []NamedPort `json:"ports,omitempty"`
+	// ExternalNode is the opaque identifier of the agent/controller responsible
+	// for additional processing or handling of this external entity.
+	ExternalNode string `json:"externalNode,omitempty"`
+}
+
+// Endpoint refers to an endpoint associated with the ExternalEntity.
+type Endpoint struct {
+	// IP associated with this endpoint.
+	IP string `json:"ip,omitempty"`
+	// Name identifies this endpoint. Could be the interface name in case of VMs.
```

Comment:
Nit: Could be the network interface name ...

Reply:
Done